pkg/controllers/management/auth/project_cluster/common.go

```diff
@@ -75,3 +80,89 @@ func reconcileResourceToNamespace(obj runtime.Object, controller string, nsListe
 		return obj, nil
 	})
 }
+
+// createMembershipRoles creates 2 cluster roles: an owner role and a member role. To be used to create project/cluster membership roles.
+func createMembershipRoles(obj runtime.Object, crClient crbacv1.ClusterRoleController) error {
+	var resourceName, resourceType string
+	var isCluster bool
+
+	switch v := obj.(type) {
+	case *apisv3.Project:
+		resourceType = apisv3.ProjectResourceName
+		resourceName = v.GetName()
+	case *apisv3.Cluster:
+		isCluster = true
+		resourceType = apisv3.ClusterResourceName
+		resourceName = v.GetName()
+	default:
+		return fmt.Errorf("cannot create membership roles for unsupported type %T", v)
+	}
+
+	ownerRef, err := newOwnerReference(obj)
```

Comment:
Does this work for `Projects`? `ClusterRoles` are non-namespaced and `Projects` are namespaced. See https://github.com/rancher/rancher/pull/48682#discussion_r1919286170